\begin{table}[h!]
\caption{Supplementary table 1.}
\label{tab:ch2_suptable1}
\resizebox{\linewidth}{!}{%
\begin{tabular}{ll}
\toprule
\multicolumn{2}{c}{\textbf{FastQ Toolkit v2.2.0}}                                       \\ \midrule
Minimum read length                             & 32                           \\
Sub-sampling                                    & FALSE                        \\
Adapter trim stringency                         & 0.9                          \\
Select respective adapters                      & TRUE                         \\
Quality trimming                                & FALSE                        \\
Poly-A/T Trimming                               & FALSE                        \\
Read Filtering                                  & FALSE                        \\
Modify Reads                                    & FALSE                        \\
Fix Format                                      & FALSE                        \\ \midrule
\multicolumn{2}{c}{\textbf{FastQC v1.0.0}}                                              \\ \midrule
Kmer Size                                       & 5                            \\
Use Conatminant Filter                          & TRUE                         \\ \midrule
\multicolumn{2}{c}{\textbf{Kraken Metagenomics v1.0.0}}                                 \\ \midrule
Host Filter                                     & TRUE RefSeqhg19              \\
Classification Database                         & MiniKraken 20141208 (latest) \\
Filter Threshold                                & 0                            \\ \midrule
\multicolumn{2}{c}{\textbf{Metaphlan v1.0.0}}                                           \\ \midrule
Sensitivity options for read-marker similarity (as descibed by BowTie2)                  & Very Sensitive \\
\multicolumn{2}{c}{\textbf{SPAdes Genome Assembler v3.9.0}}                             \\
Running Mode                                    & Error Correction \& Assembly \\
Dataset type                                    & Multi Cell                   \\
Careful Mode                                    & Disable                      \\
k-mer lengths                                   & Auto                         \\ \midrule
\multicolumn{2}{c}{\textbf{SEAR: Antibiotic Resistance v1.0.0}}                                           \\ \midrule
Read length cutoff (bases)                      & 70                           \\
Read quality score cutoff                       & 20                           \\
Read subtraction against E.coli reference genome (K12)?                                  & No             \\
Clustering stringency (express \% as a decimal) & 0.98                         \\
Annotation stringency (\% length of reference ARG sequence mapped to by sequencing reads & 40             \\\midrule
\multicolumn{2}{c}{\textbf{GENIUS Metagenomics: Know Now v1.1.0}}                                                  \\ \midrule
Can't set any settings in BaseSpace             & \\                    \midrule    
\end{tabular}%
}
\end{table}