\begin{table}[]
\caption{Microorganisms identified by conventional methods, \ac{WGS} and using shotgun metagenomics and the taxonomic classification methods in webpages (BaseSpace, Taxonomer and CosmosID).}
\label{tab:ch2_table4}
\resizebox{\linewidth}{!}{%
\begin{tabular}{@{}|l|l|l|l|lllll|@{}}
\toprule
\multirow{2}{*}{\textbf{Sample number}} &
  \multirow{2}{*}{\textbf{Culture result (CFU)$^a$}} &
  \multirow{2}{*}{\textbf{\begin{tabular}[c]{@{}l@{}}Conventional identification\\  (MALDI-TOF)\end{tabular}}} &
  \multirow{2}{*}{\textbf{\ac{WGS}-based identification}} &
  \multicolumn{5}{c|}{\textbf{Shotgun metagenomics}} \\ \cmidrule(l){5-9} 
 &
   &
   &
   &
  \multicolumn{1}{l|}{\textbf{\begin{tabular}[c]{@{}l@{}}Genius\\  (Basespace)$^c$\end{tabular}}} &
  \multicolumn{1}{l|}{\textbf{\begin{tabular}[c]{@{}l@{}}Kraken\\  (Basespace)$^c,d$\end{tabular}}} &
  \multicolumn{1}{l|}{\textbf{\begin{tabular}[c]{@{}l@{}}MetaPhlAn\\  (Basespace)$^c$\end{tabular}}} &
  \multicolumn{1}{l|}{\textbf{Taxonomer (Utah)$^b,e$}} &
  \textbf{Cosmos ID$^a$} \\ \midrule
\textbf{1} &
  \begin{tabular}[c]{@{}l@{}}10$^3$\\  10$^3$\\  10\end{tabular} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. faecium}\\  \textit{S. haemolyticus}\\  \textit{C. glabrata}\end{tabular} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. faecium}\\  \textit{S. haemolyticus}\\  -\end{tabular} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. faecium} (14.4\%)\\  \textit{S. haemolyticus} (55.8\%)\\  -\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. faecium} (25.0\%)\\  \textit{S. haemolyticus} (20.1\%)\\  -\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. faecium} (65.1\%)\\  \textit{S. haemolyticys} (30.4\%)\\  -\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. faecium} (22.9\%)\\  \textit{S. haemolyticus} (20.1\%)\\  Not identified$^*$\end{tabular}} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. faecium} (50.3\%)\\  \textit{S. haemolyticus} (22.1\%)\\  \textit{C. glabrata} (88.6\%)\end{tabular} \\ \midrule
\textbf{2} &
  \begin{tabular}[c]{@{}l@{}}103\\  1\\  Not determined\end{tabular} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. avium}\\  \textit{E. coli}\\  Anaerobes\end{tabular} &
  \begin{tabular}[c]{@{}l@{}}-$^\#$\\  -$^\#$\\  -$^\#$\end{tabular} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}Not identified$^*$\\  Not identified$^*$\\  Several species (94.0\%)\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}Not identified$^*$\\  Not identified$^*$\\  Several species (27.0\%)\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}Not identified$^*$\\  Not identified$^*$\\  Several species (54.2\%)\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}Not identified$^*$\\  Not identified$^*$\\  Several species (14.2\%)\end{tabular}} &
  \begin{tabular}[c]{@{}l@{}}Not identified$^*$\\  Not identified$^*$\\  Several species (100\%)\end{tabular} \\ \midrule
\textbf{3} &
  1 &
  \textit{S. epidermidis} &
  -\# &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (0.1\%)} &
  \multicolumn{1}{l|}{Not identified$^*$} &
  \multicolumn{1}{l|}{\textit{S. pseudintermedius} (3.4\%)} &
  Not identified$^*$ \\ \midrule
\textbf{4} &
  10$^3$ &
  \textit{S. aureus} &
  \textit{S. aureus} &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (0.3\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (8.3\%)} &
  \textit{S. aureus} (100\%) \\ \midrule
\textbf{5} &
  \begin{tabular}[c]{@{}l@{}}$\geq$ 10$^5$\\  $\geq$ 10$^5$\\  10$^3$\\  10$^3$\\  Not determined\\  10\end{tabular} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. coli}\\  \textit{K. oxytoca}\\  \textit{S. anginosus}\\  \textit{E. faecalis}\\  Anaerobes\\  \textit{C. albicans}\end{tabular} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. coli}\\  \textit{K. oxytoca}\\  -$^\#$\\  \textit{E. faecalis}\\  -$^\#$\\  -$^\#$\end{tabular} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. coli} (0.4\%)\\  Not identified$^*$\\  \textit{S. anginosus} (0.03\%)\\  \textit{E. faecalis} (0.8\%)\\  Several species (45.0\%)\\  -\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. coli} (10.2\%)\\  \textit{K. oxytoca} (0.5\%)\\  \textit{S. anginosus} (0.4\%)\\  \textit{E. faecalis} (0.3\%)\\  Several species (8.0\%)\\  -\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. coli} (7.0\%)\\  \textit{K. pneumoniae} (0.01\%)\\  \textit{S. anginosus} (0.3\%)\\  \textit{E. faecalis} (0.7\%)\\  Several species (89.1\%)\\  -\end{tabular}} &
  \multicolumn{1}{l|}{\begin{tabular}[c]{@{}l@{}}\textit{E. coli} (3.6\%)\\  \textit{K. michiganensis} (0.1\%)\\  \textit{S. anginosus} (0.1\%)\\  \textit{E. faecalis} (0.1\%)\\  Several species (60.3\%)\\  -\end{tabular}} &
  \begin{tabular}[c]{@{}l@{}}\textit{E. coli} (7.6\%)\\  \textit{K. oxytoca} (1.7\%)\\  \textit{S. anginosus} (0.09\%)\\  \textit{E. faecalis} (3.7\%)\\  Several species (86.2\%)\\  Not identified$^*$\end{tabular} \\ \midrule
\textbf{6} &
  10$^3$ &
  \textit{E. faecium} &
  \textit{E. faecium} &
  \multicolumn{1}{l|}{\textit{E. faecium} (4.2\%)} &
  \multicolumn{1}{l|}{\textit{E. faecium} (14.8\%)} &
  \multicolumn{1}{l|}{\textit{E. faecium} (5.5\%)} &
  \multicolumn{1}{l|}{\textit{E. faecium} (1.4\%)} &
  \textit{E. faecium} (4.1\%) \\ \midrule
\textbf{7} &
  10$^2$ &
  \textit{S. aureus} &
  -\# &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (93.8\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (14.2\%)} &
  \textit{S. aureus} (100\%) \\ \midrule
\textbf{8} &
  10$^3$ &
  \textit{O. intermedium} &
  \textit{O. intermedium} &
  \multicolumn{1}{l|}{\textit{O. intermedium} (100\%)} &
  \multicolumn{1}{l|}{\textit{O. nthropic} (88.9\%)} &
  \multicolumn{1}{l|}{\textit{O. intermedium} (99.8\%)} &
  \multicolumn{1}{l|}{\textit{O. intermedium} (13.1\%)} &
  \textit{O. intermedium} (49.5\%) \\ \midrule
\textbf{9} &
  10$^3$ &
  \textit{S. aureus} &
  \textit{S. aureus} &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (99.5\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (100\%)} &
  \multicolumn{1}{l|}{\textit{S. aureus} (12.7\%)} &
  \textit{S. aureus} (100\%) \\ \midrule
\textbf{10} &
  10$^3$ &
  \textit{S. marcescens} &
  -$^\#$ &
  \multicolumn{1}{l|}{\textit{S. marcescens} (32.5\%)} &
  \multicolumn{1}{l|}{\textit{S. marcescens} (94.8\%)} &
  \multicolumn{1}{l|}{\textit{Serratia} spp. (100\%)} &
  \multicolumn{1}{l|}{\textit{S. marcescens} (1.4\%)} &
  S\textit{. marscescens} (38.4\%) \\ \bottomrule
\end{tabular}%
}
\small
\item $^a$The number of colonies of a given species was estimated from the number of colonies with the same morphology on the same plate 
\item $^b$The relative abundance is calculated using total number of reads as denominator
\item $^c$The relative abundance is calculated with the total number of classified reads as denominator
\item $^d$miniKraken database was used
\itel $^e$Full Analysis mode was used 
\item $^\#$Although there was a laboratory identification, no isolates were available for \ac{WGS}
\item $^*$No reads matched that specific pathogen, not even at the genus level
\end{table}
