
\begin{acronym}[MPC]

\acro{ACEGID}{African Center of Excellence for Genomics of Infectious Diseases}
\acro{AMR}{Antimicrobial Resistance}
\acro{ASM NGS}{American Society for Microbiology Next Generation Sequencing}
\acro{ATCC}{American Type Culture Collection}
\acro{bp}{basepairs}
\acro{CAMI}{Critical Assessment of Metagenome Interpretation}
\acro{CanCOGeN}{Canadian COVID Genomics Network}
\acro{CDS}{Coding Sequence}
\acro{cgMLST}{core-genome Multilocus Sequence Typing}
\acro{cg/wg MLST}{core-genome/whole genome Multilocus Sequence Typing}
\acro{CI}{Continuous Integration}
\acro{COG-UK}{COVID-19 Genomics UK Consortium}
\acro{COVID-19}{Coronavirus disease of 2019}
\acro{CSIS}{Code Safety Inspection Service}
\acro{dBg}{De Bruijn graphs}
\acro{ddNTP}{dideoxynucleotide triphosphate}
\acro{DENV}{Dengue virus}
\acro{DNA}{Deoxyribonucleic acid}
\acro{dNTP}{deoxynucleotide triphosphate}
\acro{EBI}{European Bioinformatics Institute}
\acro{EFO}{Experimental Phenotype Ontology}
\acro{EMBL-EBI}{European Molecular Biology Laboratory's European Bioinformatics Institute}
\acro{ENA}{European Nucleotide Archive}
\acro{FAIR}{Findable Accessible Interoperable Reusable}
\acro{FS}{Filtered Set}
\acro{GAZ}{Gazetteer Ontology}
\acro{GB}{Gigabytes}
\acro{GBD}{Global Burden of Disease}
\acro{GenEpiO}{Genomic Epidemiology Ontology}
\acro{GISAID}{Global Initiative on Sharing All Influenza Data}
\acro{GPP}{Global Priority Pathogens}
\acro{HP}{Human Phenotype Ontology}
\acro{HPC}{high-performance computing}
\acro{HTS}{high-throughput sequencing}
\acro{INSDC}{International Nucleotide Sequence Database Collaboration}
\acro{INSACOG}{Indian SARS-CoV-2 Genomic Consortia}
\acro{JSON}{JavaScript Object Notation}
\acro{LIMS}{Laboratory Information Management System}
\acro{LFIA}{Lateral Flow Immunoassays}
\acro{LSA}{Longest single alignment}
\acro{MAG}{Metagenomic Assembled Genome}
\acro{MERS}{Middle East Respiratory Syndrome}
\acro{MIGS}{Minimum Information about a Genomic Sequence}
\acro{MIxS}{Minimum Information about any Sequence}
\acro{MLST}{Multilocus Sequence Typing}
\acro{MP}{Mammalian Phenotype Ontology}
\acro{NCBI}{National Center for Biotechnology Information}
\acro{NCBITaxon}{NCBI Taxonomy Ontology}
\acro{NCIT}{National Cancer Institute Thesaurus}
\acro{NCR}{non-coding region}
\acro{NIST}{National Institute of Standards and Technology}
\acro{OBI}{Ontology for Biological Investigations}
\acro{OBO Foundry}{Open Biological and Biomedical Ontology Foundry}
\acro{OLC}{Overlap-Layout Consensus}
\acro{ONT}{Oxford Nanopore Technologies}
\acro{OTUs}{Operational Taxonomic Units}
\acro{PacBio}{Pacific Biosciences}
\acro{PCR}{Polymerase Chain Reaction}
\acro{PFGE}{Pulse Field Gel Electrophoresis}
\acro{PHA4GE}{Public Health Alliance for Genomic Epidemiology}
\acro{PHSS}{Public Health Surveillance Systems}
\acro{Pls}{Phred-like score}
\acro{PMC}{PubMed Central\textsuperscript{\small\textregistered}}
\acro{QC}{quality control}
\acro{qPCR}{Real-Time Quantitative PCR}
\acro{RNA}{Ribonucleic acid}
\acro{rRNA}{ribossomal RNA}
\acro{RT-PCR}{Reverse Transcription Polymerase Chain Reaction}
\acro{SANBI}{South African National Bioinformatics Institute}
\acro{SARS-CoV-2}{Acute  Respiratory  Syndrome  Coronavirus  2}
\acro{SMg}{Shotgun Metagenomics}
\acro{SMRT}{Single Molecule Real-Time Sequencing}
\acro{SNP}{Single Nucleotide Polymorphism}
\acro{SOP}{standard operating procedure}
\acro{SPHERES}{SARS-CoV-2 Sequencing for Public Health Emergency Response, Epidemiology and Surveillance}
\acro{SRA}{Sequence Read Archive}
\acro{STEC}{Shiga toxin-producing \textit{Escherichia coli}}
\acro{UBERON}{Uber-Anatomy Ontology}
\acro{URL}{Uniform Resource Locator}
\acro{UO}{Unit Ontology}
\acro{VCS}{Version control systems}
\acro{WGS}{Whole Genome Sequencing}
\acro{WHO}{World Health Organization}

\end{acronym}
