\mbox{}\\
\vspace{8cm}

This chapter is a reproduction of the following publication:

E. J. Griffiths, R. E. Timme, C. I. Mendes, A. J. Page, N. Alikhan, D. Fornika, F. Maguire, J. Campos, D. Park, I. B. Olawoye, P. E. Oluniyi, D. Anderson, A. Christoffels, A. G. da Silva, R. Cameron, D. Dooley, L. S. Katz, A. Black, I. Karsch-Mizrachi, T. Barrett, A. Johnston, T. R. Connor, S. M. Nicholls, A. A. Witney, G. H. Tyson, S. H. Tausch, A. R. Raphenya, B. Alcock, D. M. Aanensen, E. Hodcroft, W. W. L. Hsiao, A. T. R. Vasconcelos, D. R. MacCannell on behalf of the Public Health Alliance for Genomic Epidemiology (PHA4GE) consortium.
Future-proofing and maximizing the utility of metadata: The PHA4GE SARS-CoV-2 contextual data specification package. GigaScience, Volume 11, giac003 (2022). DOI: \url{https://doi.org/10.1093/gigascience/giac003}

As mentioned in Chapter \ref{ch:introduction}, the \ac{SARS-CoV-2} has brought a new meaning to genomic surveillance, with currently, over 8 million complete viral sequences are available at GISAID\footnote{\url{https://www.gisaid.org/}}, being one of the most highly sequenced genomes of any organism on the planet. 
This richness in genomic information has been basal to identifying new variants of risk and new variants of concern with a myriad of different origins, identifying routes of transmission across borders, including the identification of "super-spreaders" events, and informing infection control measures. 

Despite this richness in genomic information, the same is not observed for the metadata that accompanies it. As described in Chapter \ref{ch:paper6}, a standardised output specification was conceived, this time applied to SARS-CoV-2 contextual data based on harmonisable, publicly available community standards. This is implemented through a collection template, as well as a variety of protocols and tools to support both the harmonisation and submission of sequence data and contextual information to public biorepositories.

My contribution to this publication included the development of the SARS-CoV-2 contextual data specification package, including it's conversion and availability in a machine-applicable JSON format. I've also maintain the public repository where the data specification package is hosted\footnote{\url{https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/}} Additionally, I've also contributed to the manuscript production and editing.


\cleardoublepage 

\begin{center}
\large
\textbf{Future-proofing and maximizing the utility of metadata: The PHA4GE SARS-CoV-2 contextual data specification package}
\end{center}

Emma J. Griffiths$^1$, 
Ruth E. Timme$^2$,
Catarina I. Mende$^3$,
Andrew J Page$^4$,
Nabil-Fareed Alikha$^4$,
Dan Fornika$^5$,
Finlay Maguire$^6$,
Josefina Campos$^7$,
Daniel Park$^8$,
Idowu B. Olawoy$^{9,10}$,
Paul E. Oluniy$^{9,10}$,
Dominique Anderson$^{11}$,
Alan Christoffel$^{11}$,
Anders Gonçalves da Silva$^{12}$,
Rhiannon Cameron$^1$,
Damion Dooley$^1$,
Lee S. Katz$^{13}$,
Allison Black$^{14}$,
Ilene Karsch-Mizrach$^{15}$,
Tanya Barret$^{15}$,
Anjanette Johnston$^{15}$,
Thomas R. Connor$^{16,17}$,
Samuel M. Nicholls$^{18}$,
Adam A. Witney$^{19}$,
Gregory H. Tyson$^{20}$,
Simon H. Tausch$^{21}$,
Amogelang R. Raphenya$^{22}$,
Brian Alcock$^{22}$,
David M. Aanensen$^{23,24}$,
Emma Hodcrof$^{25}$,
William W. L. Hsiao$^{1,5,26}$,
Ana Tereza R Vasconcelos$^{27}$,
Duncan R MacCannel$^{28}$,
on behalf of the Public Health Alliance for Genomic Epidemiology (PHA4GE) consortium

$^1$  Faculty of Health Sciences, Simon Fraser University, Burnaby, British Columbia,Canada;

$^2$ Center for Food Safety and Applied Nutrition, U.S. Food and Drug Administration, College Park, MD,USA; 

$^3$ Instituto de Microbiologia, Instituto de Medicina Molecular, Faculdade de Medicina, Universidade de Lisboa, Portugal;

$^4$ Quadram Institute Bioscience, Norwich, Norfolk, UK;

$^5$ BC Centre for Disease Control Public Health Laboratory, Vancouver, Canada;

$^6$ Faculty of Computer Science, Dalhousie University, Halifax, Canada;

$^7$ INEI-ANLIS "Dr Carlos G. Malbrán", Buenos Aires, Argentina;

$^8$ The Broad Institute of MIT and Harvard, Cambridge, MA, USA;

$^9$ African Center of Excellence for Genomics of Infectious Diseases (ACEGID), Redeemer's University, Ede, Osun State, Nigeria;

$^{10}$ Department of Biological Sciences, College of Natural Sciences, Redeemer's University, Ede, Osun State, Nigeria;

$^{11}$ South African Medical Research Council Bioinformatics Unit, South African National Bioinformatics Institute, University of the Western Cape, Bellville, South Africa;

$^{12}$  Microbiological Diagnostic Unit Public Health Laboratory, The Peter Doherty Institute for Infection and Immunity, The University of Melbourne, Melbourne, Victoria, Australia;

$^{13}$ Center for Food Safety, University of Georgia, Georgia, USA;

$^{14}$ Department of Epidemiology, University of Washington, Washington, USA;

$^{15}$ National Center for Biotechnology Information, National Library of Medicine, National Institutes of Health, Bethesda, MD, USA;

$^{16}$ Organisms and Environment Division, School of Biosciences, Cardiff University, Cardiff, Wales, UK;

$^{17}$ Public Health Wales, University Hospital of Wales, Cardiff, UK;

$^{18}$ University of Birmingham, Birmingham, UK;

$^{19}$ Institute for Infection and Immunity, St George’s, University of London, London, UK;

$^{20}$ Center for Veterinary Medicine, U.S. Food and Drug Administration, Laurel, Maryland, USA;

$^{21}$ Department of Biological Safety, German Federal Institute for Risk Assessment, Berlin, Germany;

$^{22}$ Department of Biochemistry and Biomedical Sciences and the Michael G. DeGroote Institute forInfectious Disease Research, McMaster University, Hamilton, Ontario, Canada;

$^{23}$ Centre for Genomic Pathogen Surveillance, Wellcome Genome Campus, Cambridge, UK;

$^{24}$ The Big Data Institute, Li Ka Shing Centre for Health Information and Discovery, Nuffield Department of Medicine, University of Oxford, Oxford, UK;

$^{25}$ Biozentrum, University of Basel, Basel, Switzerland \& Swiss Institute of Bioinformatics, Lausanne,Switzerland;

$^{26}$Department of Pathology and Laboratory Medicine, University of British Columbia, Vancouver, Canada;

$^{27}$ Bioinformatics Laboratory National Laboratory of Scientific Computation LNCC/MCTI, Rio de Janeiro, Brazil;

$^{28}$ National Center for Emerging and Zoonotic Infectious Diseases, Centers for Disease Control and Prevention, Georgia, USA

\section{Abstract} \label{sec:ch7_abstract}

The Public Health Alliance for Genomic Epidemiology (PHA4GE) (https://pha4ge.org) is a global coalition that is actively working to establish consensus standards, document and share best practices, improve the availability of critical bioinformatics tools and resources, and advocate for greater openness, interoperability, accessibility and reproducibility in public health microbial bioinformatics. In the face of the current pandemic, PHA4GE has identified a need for a fit-for-purpose, open-source SARS-CoV-2 contextual data standard. As such, we have developed a SARS-CoV-2 contextual data specification package based on harmonisable, publicly available community standards. The specification can be implemented via a collection template, as well as an array of protocols and tools to support both the harmonisation and submission of sequence data and contextual information to public biorepositories. Well-structured, rich contextual data adds value, promotes reuse, and enables aggregation and integration of disparate data sets. Adoption of the proposed standard and practices will better enable interoperability between datasets and systems, improve the consistency and utility of generated data, and ultimately facilitate novel insights and discoveries in SARS-CoV-2 and COVID-19. The package is now supported by the National Center for Biotechnology (NCBI)'s BioSample database.

\section{Findings} \label{sec:ch7_findings}

\subsection{The importance of contextual data for interpreting SARS-CoV-2 sequences}

First identified in late 2019 in Wuhan, China, the SARS-CoV-2 virus has now spread to virtually every country and territory in the world, resulting in millions of confirmed cases, and deaths, globally \cite{world_health_organization_coronavirus_nodate, dong_interactive_2020}. Understanding, monitoring and preventing transmission, as well as the development of vaccines and effective therapeutic options, have been primary goals of the public health response to SARS-CoV-2. 

Tracking the spread and evolution of the virus at global, national and local scales has been aided by the analysis of viral genome sequence data alongside SARS-CoV-2 epidemiology. Large scale sequencing efforts are often formalised as consortia across the world, including the COG-UK in the UK \cite{covid-19_genomics_uk_cog-uk_consortiumcontactcogconsortiumuk_integrated_2020}, SPHERES in the USA \cite{cdc_cases_2020}, CanCOGeN in Canada \cite{cancogen_genome_canada_cancogen_nodate}, Latin American Genomics SARS-CoV-2 Network \cite{pan_american_health_organization_laboratory_nodate, candido_evolution_2020}, 2019nCoVR in China \cite{zhao_2019_2020}, the South Africa NGS Genomic Surveillance Network \cite{network_for_genomic_surveillance_south_africa_ngs-sa_nodate}, AusTrakka in Australia and New Zealand \cite{communicable_diseases_genomics_network_austrakka_nodate}, and INSACOG in India \cite{government_of_india_indian_nodate}. In addition to these initiatives, many agencies, universities and hospital laboratories around the world are also sequencing and sharing sequence data at an unprecedented pace. Deposition of these sequences into public repositories such as the Global Initiative on Sharing All Influenza Data (GISAID) and the International Nucleotide Sequence Database Collaboration (INSDC) has enabled rapid global sharing of data \cite{shu_gisaid_2017,karsch-mizrachi_international_2018}. At the time of writing, 174 countries had undertaken open sequencing initiatives (GISAID accessed 2021-06-23) depositing 2,057,675 sequences which are being reused and analysed on a massive scale. The open data sharing paradigm has had tremendous success in the genomic epidemiology of foodborne pathogens \cite{allard_practical_2016,kubota_pulsenet_2019}, and has the potential to reveal a deeper understanding of SARS-CoV-2 origin, pathogenicity, and basic biology when submissions from environmental samples and wild hosts are included alongside human clinical samples \cite{cook_integrating_2020}.

SARS-CoV-2 sequencing, analysis, and open sharing have played a crucial role in a number of developments during the pandemic, such as dispelling misinformation about the origins of the virus \cite{andersen_proximal_2020}, the identification and surveillance of variants of concern \cite{gupta_will_2021}, \cite{public_health_england_sars-cov-2_nodate}, the improvement of diagnostic performance and rapid testing \cite{los_alamos_national_laboratory_silico_nodate, kuchinski_mutations_2022, ganguli_rapid_2020}, and the development of vaccines which are currently being distributed in the largest global vaccination program the world has ever seen \cite{world_health_organization_covid-19_nodate}. Viral genomic sequences are also being used to understand transmission and reinfection events \cite{tillett_genomic_2021} as well to monitor the prevalence and diversity of lineages during different exposure events and in different settings e.g. animal reservoirs \cite{oude_munnink_transmission_2021}, long-term care facilities \cite{lai_covid-19_2020, aggarwal_role_2020, murti_investigation_2021}, healthcare and other work sites \cite{dyal_covid-19_2020, gunther_sarscov2_2020, taylor_serial_2020, loconsole_investigation_2021, frampton_genomic_2021}, conferences and other public gatherings \cite{da_silva_filipe_genomic_2021}, as well before and after public health responses (e.g. border controls and travel restrictions, lockdowns and quarantines, vaccination, etc.), through successive waves of infections \cite{oude_munnink_rapid_2020, du_plessis_establishment_2021, githinji_tracking_2020, meredith_rapid_2020, zhang_analysis_2020, long_molecular_2020, geoghegan_genomic_2020, seemann_tracking_2020, mclaughlin_early_2021, fauver_coast--coast_2020, knock_key_2021, lane_genomics-informed_2021}. However, it is critical to note that public health sequence data is of limited value without accompanying contextual metadata. 

Contextual data consists of sample metadata (e.g., collection date, sample type, geographical location of sample collection), as well as laboratory (e.g., date and location testing, cycle threshold (CT) values), clinical outcomes (e.g., hospitalisation, death, recovery), epidemiological (e.g., age, gender, exposures, vaccination status) and methods (e.g., sampling, sequencing, bioinformatics) that enable the interpretation sequence data (e.g., previous examples). High-quality contextual data is also crucial for quality control. For example, detecting systematic batch effect errors related to certain sequencing centres and methods can help evaluate which variants represent real, circulating viruses, as opposed to artefacts of sample handling or sequencing which may arise due to different aspects experimental design, laboratory procedures, bioinformatics processing, and applied quality control thresholds \cite{de_maio_issues_2020, rayko_quality_2020, poon_recurrent_2005}. 

Good data stewardship practices are not only critical for auditability and reproducibility, but for posterity - documenting critical information about samples, methods, risk factors and outcomes etc., can help future-proof information used to build a roadmap for dealing with future public health crises. Contextual data, however, is often collected on a project-specific basis according to local needs and reporting requirements which results in the collection of different data types at different levels of granularity, with different meanings and implicit bias of variables and attributes. Furthermore, the information is often collected as free text, or if structured, according to organisation or initiative-specific data dictionaries, using different fields, terms, formats, abbreviations, and jargon.

The variability in the way information is encoded in private databases tends to propagate to public repositories, which makes the information more difficult to interpret and to use. There are different existing standards that can be used to structure contextual data, like minimum information checklists (MIxS \cite{yilmaz_minimum_2011}, MIGS \cite{field_minimum_2008}, the NIAID/BRC Project and Sample Application Standard \cite{dugan_standardized_2014}) and various interoperable ontologies (OBO Foundry \cite{smith_obo_2007}), which make information easier to aggregate and reuse for different types of analyses. However, these attribute packages and metadata standards developed by different organisations are usually scoped to cover as many use cases and pathogens as possible, and as such, can include fields of information not applicable to SARS-CoV-2, or that may be subject to privacy concerns, or exclude fields commonly used in public health surveillance and investigations. As different types of contextual data are subject to different ethical, practical and privacy concerns, not all components of existing standards are immediately or widely collectable and shareable. As a result, the range of generic metadata standards being applied to SARS-CoV-2 data presents challenges for data harmonization \cite{schriml_covid-19_2020} and analysis critical for fighting the disease and ending the pandemic. 

In light of these challenges, PHA4GE has identified a need for a fit-for-purpose, open-source SARS-CoV-2 contextual data specification which can be used to consistently structure information as part of good data management practices and for data sharing with trusted partners and/or public repositories. The specification was developed by consensus among domain experts, and incorporates existing community standards with an emphasis on SARS-CoV-2 public health needs and ensuring privacy while maximising information content and interoperability across datasets and databases to better enable analyses to fight COVID-19. The specification package also contains a number of accompanying materials such as standard operating procedures, tools, a reference guide, and repository submission protocols (protocols.io) to help put the standard into practice.  

\subsection{SARS-CoV-2 Contextual Data Specification: The Framework}

The purpose of the PHA4GE SARS-CoV-2 specification is to provide a mechanism for consistent structure, collection and formatting of fields and values containing SARS-CoV-2 contextual data pertaining to clinical, animal, and environmental samples. We emphasise that the purpose of this specification is not to force data sharing, but rather to provide a framework to structure data consistently across disparate laboratory and epidemiological databases so that they can be harmonised for different uses (Figure \ref{fig:chap7_figure_1}). Data sharing is just one use case and can involve sharing between divisions within a single agency, sharing between partners based on memorandums of understanding, or submission to public repositories. 

The PHA4GE SARS-CoV-2 contextual data specification was created through broad consultation with representatives from public health laboratories, research institutes and universities in 11 countries (Argentina, Australia, Brazil, Canada, Germany, Nigeria, Portugal, South Africa, Switzerland, the United Kingdom, the United States of America) who are involved with the SARS-CoV-2 genome sequencing and analysis efforts at various scales. Based on this consultation and consensus, the specification contains different fields covering a wide array of data types described in Box 1 (Figure \ref{fig:chap7_figure_1}). The specification attempts to harmonise different data standards (INSDC, GISAID, MIxS, MIGS, Sample Application Standard) by reusing fields or mapping to fields, as much as possible. As PHA4GE embraces FAIR data stewardship principles (Findability, Accessibility, Interoperability and Reuse of digital assets), we strived to implement FAIR principles in the design and implementation of the specification for data management and data sharing. At their core, these principles emphasise machine-actionability and consistency of data, and are critical for dealing with the volume and complexity of genomic sequence and contextual data. Principles of FAIR data stewardship that have been implemented include improving machine-actionability of data by using a formal, accessible, shared, and broadly applicable language for knowledge representation, reusing existing standards and ontology-based vocabulary to increase interoperability, providing a data usage license, capturing data provenance, and making all resources open, free and widely accessible.

\begin{figure*}[h!]
\centering
\includegraphics[width=\textwidth]{figures/chapter 7/giac003fig1.jpeg}
\caption{Contextual data flow. Contextual data can be captured and structured using the PHA4GE specification so that they can be more easily harmonised across different data sources and providers. Different subsets of the harmonised data can be (i) shared with public repositories, e.g., GISAID and INSDC; (ii) shared with trusted partners, e.g., national sequencing consortia, public health partners; and (iii) kept private and retained locally with the potential for sharing in the future for particular surveillance or research activities. While fields have been colour-coded in the template to indicate whether they are considered “required,” “strongly recommended,” or “optional,” how the specification is implemented and whether any of the data are shared is ultimately at the discretion of the user. Box 1 describes the information types covered in the full specification.}
\label{fig:chap7_figure_1}
\end{figure*}

The versioned specification is available as a contextual data collection template (.xlsx) and in machine-amenable JSON format from GitHub (v 3.0.0 - https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification) \cite{public_health_alliance_for_genomic_epidemiology_sars-cov-2-contextual-data-specification_nodate}. The collection template also offers standardised terms for a number of fields in the form of pick lists. The fields are colour-coded to indicate required (yellow), strongly recommended (purple) or optional status (white). Fields useful for surveillance were prioritised as required. Formats for data elements like dates are also prescribed according to international standards (e.g., dates should be formatted according to ISO8601).

The template is also supported by several materials such as term and field-level Reference Guides (available as tabs in the collection template Excel workbook), which provides definitions, data entry guidance and examples of usage \cite{public_health_alliance_for_genomic_epidemiology_sars-cov-2-contextual-data-specification_nodate}. The field-level Reference Guide also provides mapping of PHA4GE fields to existing contextual data standards, highlighting public health and SARS-CoV-2-specific fields that were missing as well as fields in those other standards that were considered out of scope. 

The Open Biological and Biomedical Ontology (OBO) Foundry is a community of researchers that use a prescribed set of principles and practices to develop a wide range of interoperable ontologies focused on the life sciences \cite{the_obo_foundry_obo_nodate}. Fields and terms in the specification have been mapped to existing OBO Foundry ontology terms, and where required, new ontology terms have been developed and are being made available in different application and domain-specific ontologies within The Foundry (see Table \ref{tab:ch7_table1} for a list of source ontologies). As of version 3.0.0 and beyond, terms in pick lists provided in the collection template are presented with corresponding ontology identifiers in the format “Label [ontology ID]” e.g., Blood [UBERON:0000178]. Axioms and additional cross references to ontologies and existing standards are actively being developed in collaboration with community developers. We anticipate that our contributions to these freely available, open-source resources will be of use to the COVID-19 research community.

\input{tables/chapter_7/table1}

Protocols have also been created and are openly available on protocols.io \cite{public_health_alliance_for_genomic_epidemiology_pha4ge_nodate}, including a curation Standard Operating Procedure (SOP) containing instructions for using the collection template as well as guidance for a number of privacy and practical concerns. A series of versioned SARS-CoV-2 sequence and contextual data submission protocols and accompanying instructional videos for how to prepare submissions and navigate through the various submission portals for GISAID, NCBI and EMBL-EBI are also provided.

A mapping file indicating which PHA4GE fields correspond to contextual data elements recommended by the World Health Organization has been provided to help data providers comply with international guidance \cite{world_health_organization_guidance_nodate}. This mapping file also includes tabs indicating which PHA4GE fields correspond to those found in different repository submission forms to facilitate data transformations for submissions. Such transformations can be automated using a contextual data harmonization application called the DataHarmonizer \cite{hsiao_public_health_bioinformatics_lab_dataharmonizer_2022}. PHA4GE has worked with the developers of the DataHarmonizer to offer the PHA4GE standard as a template in the tool (Gill et al, in preparation). Users can standardize and validate entered data and export it as GISAID and NCBI-ready submission forms (BioSample, SRA, GenBank and GenBank source modifier forms). It should be noted that other excellent contextual data transformation tools have been developed by the community, such as METAGENOTE, multiSub, and a GISAID-to-ENA conversion script \cite{noauthor_metagenote_nodate, haeussler_multisub_2022, noauthor_ena-content-dataflowscriptsgisaid_to_ena_nodate}.

A table outlining the different specification package materials can be found in Table \ref{tab:ch7_table2}.

\input{tables/chapter_7/table2}

\subsection{Getting Started - How To Use The Standard}

In designing the specification we first considered the goals of data collection and harmonization. Consulted stakeholders felt that the primary priority of standardising data should be improved support for SARS-CoV-2 genomic surveillance activities and the submission of sequence data and minimal metadata to public repositories. The two most important attributes for tracking transmission from pathogen genomic data are temporal information describing when a sample was collected and spatial information describing where a virus was sampled.

Comparisons of minimal contextual data requirements across different national sequencing efforts, as well as submission requirements for INSDC and GISAID databases, yielded a minimal set of 14 fields which have been annotated as “required” in the specification (colour-coded yellow in the collection template). The required fields, corresponding definitions, and guidance notes are described in Table \ref{tab:ch7_table3}. A number of other fields have been annotated as “strongly recommended” (colour-coded purple in the collection template) for capturing sample collection and processing methods, critical epidemiological information about the host, and acknowledging scientific contributions. Fields colour-coded white are considered optional. 

\input{tables/chapter_7/table3}

As many contextual data fields are stored in different locations and databases (e.g., LIMS, epidemiology case report forms and databases), a benefit of implementing the PHA4GE collection template is that it enables the capture of these different pieces of information in one place. The collection template also offers picklists for a variety of fields e.g., a curated INSDC country list for “geo\_loc name (country)”, the standardised name of the virus under the “organism” field (i.e., Severe acute respiratory coronavirus 2), and a multitude of standardised terms for sample types (anatomical materials and sites, environmental materials and sites, collection devices and methods). The “purpose of sequencing” field provides standardised tags which can be used to highlight sampling strategy criteria (e.g., baseline surveillance (random sampling), targeted sequencing (non-random sampling), which are very important for understanding bias when interpreting patterns in sequence data. The picklists provided are neither exhaustive, nor comprehensive, but have been curated from current literature representing active sampling and surveillance activities.

If a pick list is missing standardised terms of interest, the reference guide also provides links to different ontology look-up services enabling users to identify additional standardised terms. The reference guide provides definitions for the fields, additional guidance regarding the structure of the values in the field, and any suggestions for addressing issues pertaining to privacy and identifiability. The curation SOP provides users with step-by-step instructions for populating the template, looking up standardised terms, and how best to structure sample descriptions. The SOP also highlights a number of ethical, practical, and privacy considerations for data sharing.


\subsection{Implementation of the PHA4GE specification around the world}

The amount of, and manner in which the specification is implemented is ultimately at the discretion of the user. To date, versions of the specification are being implemented in the CanCOGeN (Canada) and SPHERES (USA) SARS-CoV-2 sequencing initiatives, the AusTrakka (Australia and New Zealand) data sharing platform \cite{world_health_organization_coronavirus_nodate, dong_interactive_2020, covid-19_genomics_uk_cog-uk_consortiumcontactcogconsortiumuk_integrated_2020}, by the Global Emerging Pathogens Treatment Consortium (Africa) \cite{noauthor_get_nodate}, the African Centre of Excellence for Genomics of Infectious Diseases (ACEGID) in Nigeria \cite{noauthor_acegid_nodate}, the Baobab LIMS \cite{noauthor_welcome_nodate} at the South African National Bioinformatics Institute (SANBI) \cite{noauthor_sanbi_nodate}, and the Latin American Genomics Network \cite{noauthor_covid-19_nodate}. 

Canada is implementing a version of the PHA4GE specification to harmonise contextual data across all data providers for national SARS-CoV-2 surveillance [5]. Harmonised contextual information is provided by different jurisdictions, and stored in the national genomics surveillance database at the Public Health Agency of Canada’s National Microbiology Laboratory. A worked example is provided to demonstrate how free text information can be structured according to the specification, and how subsets of the contextual data can be shared according to jurisdictional policies (Figure \ref{fig:chap7_figure_2}). 

\begin{figure*}[]
\centering
\includegraphics[width=\textwidth]{figures/chapter 7/giac003fig2.jpeg}
\caption{The PHA4GE specification is being implemented in CanCOGeN to harmonise contextual data across jurisdictions. (A) CanCOGeN is Canada's SARS-CoV-2 national genomic surveillance initiative. Canada has a decentralised health system, with one federal and 13 provincial/territorial public health jurisdictions. Provinces/Territories have authority over how data are collected, stored, and shared. Every Canadian public health jurisdiction uses different collection instruments (e.g., case report forms), different data management systems, and different pipelines and software to perform bioinformatic analyses. Provinces/Territories share sequencing data and accompanying contextual data with the National Microbiology Lab's national SARS-CoV-2 genomics database (starred) according to a version of the PHA4GE specification for national surveillance activities. (B) Excerpts from two different province-specific case collection forms. Sample type information is collected in data collection instruments using different fields, different terms, at different levels of granularity, using abbreviations and formats. BAL: bronchoalveolar lavage; NPS: nasopharyngeal swab; UTM: universal transport medium. (C) An anonymised example of how the standard consistently structures contextual information and how it is being used for data sharing. The contextual data specification provides a wide variety of fields and pick lists of terms. In the example, the full set of standardised information shown would be shared by the province with the national database. Standardised information in boldface would be shared with public repositories; however select data elements (underscored) would be withheld according to jurisdictional data sharing policies. The specification enables users to harmonise and integrate data provenance, sampling strategy criteria, epidemiological information, and methods.}
\label{fig:chap7_figure_2}
\end{figure*}

While the primary use case of the specification is for clinical sequencing, the sample collection fields have been developed to enable capture of information for a wide range of sample types, including environmental samples (e.g., swabs of hospital equipment and patient rooms, wastewater samples) and non-human hosts (e.g., wildlife, agricultural animal samples).

\subsection{Submitting Data to Public Sequence Repositories }

Most existing SARS-CoV-2 sequences have only been deposited in GISAID, with a proportion of submitters also depositing matching raw read data in the INSDC (i.e., National Center for Biotechnology Information (NCBI), European Molecular Biology Laboratory-European Bioinformatics Institute (EMBL-EBI) and DNA Data Bank of Japan (DDBJ)). While consensus genomes are widely deposited and used for public surveillance purposes, raw read data is critical for comparing methods, assessing reproducibility, as well as identifying minor variants. Linkage of contextual data to consensus sequences as well as raw data in public repositories is vital. 

Within the INSDC, the contextual data is stored as accessioned BioSamples \cite{barrett_bioproject_2012} with a consistent set of attribute names and standardised values. BioSamples add value, promote reuse, and enable interoperability of data submitted from laboratories that may only be connected by following the same metadata standard. The INSDC databases have until recently provided a generic pathogen metadata template for the BioSample that is heavily utilised for bacterial genomic surveillance \cite{noauthor_home_nodate}. GISAID uses a different format and data structure for associating metadata primarily for influenza surveillance and now extended to include SARS-CoV-2. The ENA provides a virus metadata checklist (ENA virus pathogen reporting standard checklist) developed as part of the COMPARE project \cite{compare_home_nodate}, which is very similar to the GISAID submission requirements.

Building off of these existing standards, a metadata specification for SARS-CoV-2 genomic surveillance was developed that is broad enough for internal laboratory use while providing mechanisms for mapping/transforming standardised contextual data for public release to INSDC and GISAID. Recently, PHA4GE worked with NCBI to develop a dedicated SARS-CoV-2 BioSample submission package in the NCBI Submission Portal, which incorporates many fields from the PHA4GE standard \cite{staff_dedicated_2021}. The Genomics Standards Consortium will also align its forthcoming “MIxS for SARS-CoV-2” package with this specification. EMBL-EBI will also offer the PHA4GE standard to submitters as one of its validated checklists. Taken together, the PHA4GE specification has already had widespread impact on contextual information data structures around the world.

The detailed mapping of PHA4GE fields to public repository submission requirements, as well as guidance and advice, are available as supporting documents (see Table \ref{tab:ch7_table1}). We have also provided detailed protocols for data submission to the three participating repositories, GenBank/SRA (NCBI), ENA (EMBL-EBI), and GISAID. An overview of how the PHA4GE specification is integrated into public repository submissions is presented in Figure \ref{fig:chap7_figure_3}. PHA4GE recommendations for FAIR SARS-CoV-2 data submissions are as follows:

\begin{enumerate}
    \item submit raw sequencing data and assembled/consensus genomes to INSDC and GISAID when permitted by jurisdictional data sharing policies
    \item create a BioSample record when submitting to the INSDC using the PHA4GE guidance, populating the mandatory and recommended fields where possible
    \item curate public records (sequence data and contextual data), updating them when subsequent information becomes available or retracting if/when records become untrustworthy. 
\end{enumerate}

\begin{figure*}[h!]
\centering
\includegraphics[width=\textwidth]{figures/chapter 7/giac003fig3.jpeg}
\caption{Overview of how the PHA4GE SARS-CoV-2 contextual data specification can be integrated into public repository submission. The PHA4GE collection template provides a one-stop shop for different data types that are important for global surveillance. The protocols provided as part of the specification package describe how PHA4GE fields can be mapped to different repository submission forms. Consensus sequences (FASTA), accompanied by a subset of PHA4GE fields, can be submitted to the GISAID EpiCoV database (A). Consensus sequences (FASTA) (B) as well as raw/processed data (FASTQ, BAM) (C, D) can be submitted to INSDC databases (e.g., GenBank, SRA) with different subsets of PHA4GE fields as part of a BioSample record. BioSamples are propagated throughout INSDC databases.}
\label{fig:chap7_figure_3}
\end{figure*}

The specification has been used to submit standardised contextual data to different repositories by labs and sequencing initiatives globally. A selection of accession numbers for submissions to different repositories is provided below (Table \ref{tab:ch7_table4}).

\input{tables/chapter_7/table4}

\section{Conclusion}

The collective response to the SARS-CoV-2 pandemic has resulted in an unprecedented deployment of genomic surveillance worldwide, bringing together public health agencies, academic research institutions, and industry partners. This unified action provides opportunities to more effectively understand and respond to the pandemic. Yet it also provides an enormous challenge, as realising the full potential of this opportunity will require standardisation and harmonization of data collection across these partners. With our SARS-CoV-2 metadata specification we have endeavoured to create a mechanism for promoting consistent, standardised contextual data collection that can be applied broadly. We envision that given the increased uptake, this specification will improve the consistency of collected data, making information reusable by agencies as they continue working towards an increased understanding of SARS-CoV-2 epidemiology and biology, and harmonising them such that community-based data sharing efforts are not excessively burdened. We anticipate that the experience and lessons learned creating the specification package for SARS-CoV-2 will better enable the rapid development and deployment of pathogen-specific standards for public health pathogen genomic surveillance in the future.

\section{Methods}

The PHA4GE SARS-CoV-2 data specification was developed by first comparing existing metadata standards (e.g., MIxS/MIGS, the NIAID/BRC Sample Application Standard), various sequence repository submission requirements (e.g., GISAID, INSDC), as well as national and international case report forms. 

A gap analysis was performed to identify SARS-CoV-2 public health surveillance data elements that were missing in available standards. Fields in existing standards that were deemed to be out of scope were excluded from the specification. Terms for pick lists were sourced from public health documents, the literature, and when available, various interoperable ontologies (OBO Foundry). The fields and terms from the gap analysis were structured in the collection template (.xlsx). Field definitions, guidance for use, examples and mappings to various standards were developed as part of the Reference Guides provided in separate tabs in the template workbook. Vocabulary lists were also provided in a separate tab in the template workbook to enable validation, and to enable users to add terms to pick lists as needed, according to instructions provided in the curation SOP. The specification was also encoded as a JavaScript Object Notation (JSON) file. 

The specification was reviewed by public health, bioinformaticians and data standards experts from different public health agencies, research institutes and sequencing consortia and adapted according to feedback. Upon request by community members, versioned protocols for public repository submission were created and deposited in protocols.io.

The first version of the specification was made publicly available in August 2020 with a CC-BY 4.0 International attribution license. Iterative improvements were made to a development branch of the specification over the next 10 months as the pandemic evolved, and in response to user feedback and requests. The second major release (2.0) was made publicly available in May 2021. A third major release (3.0) including ontology mappings and the term-level reference guide was made publicly available in December 2021. The PHA4GE template was incorporated into the contextual data harmonization, validation and transformation tool called The DataHarmonizer through a collaborative effort with the Hsiao Public Health Bioinformatics Laboratory (Simon Fraser University). Details regarding DataHarmonizer development can be found elsewhere (manuscript in preparation).

\section{Availability and Requirements}

The software used in this study is available on GitHub. 

\textbf{Project name:} SARS-CoV-2-Contextual-Data-Specification

\textbf{Project home page:} https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification

\textbf{Operating system:} Platform independent

\textbf{Programming language:} Not applicable

\textbf{Other requirements:} xlsx-compatible spreadsheet software

\textbf{License: }CC-BY 4.0 International

\textbf{RRID:} SCR\_021378

\textbf{biotools:} pha4ge\_sars-cov-2\_contextual\_data\_specification

\section{Declarations}

\subsection{Ethics approval and consent to participate}

Not applicable.

\subsection{Consent for publication}

Not applicable.


\subsection{Competing interests}

The authors declare that they have no competing interests.

\section{Funding}

We wish to thank the Bill & Melinda Gates Foundation for supporting the establishment and work of the PHA4GE consortium. AJP and NFA gratefully acknowledge the support of the Biotechnology and Biological Sciences Research Council (BBSRC); and were supported by the Quadram Institute Bioscience BBSRC funded Core Capability Grant (project number BB/CCG1860/1) and the BBSRC Institute Strategic Programme Microbes in the Food Chain BB/R012504/1 and its constituent project BBS/E/F/000PR10352. FM was supported by a Donald Hill Family Fellowship in Computer Science. CIM was supported by the Fundação para a Ciência e Tecnologia (grant SFRH/BD/129483/2017). Work by EJG, RC, DD, and WWLH was funded by a Genome Canada Bioinformatics and Computational Biology 2017 Grant #286GET and a Genome Canada CanCOGeN grant E09CMA. The work of IKM, TB, and AJ was supported by the Intramural Research Program of the National Library of Medicine, National Institutes of Health. 

\section{Authors' contributions}

EJG: Conceptualization, Methodology, Investigation, Software, Visualization, Writing - Original Draft Preparation, Validation, Supervision; RET: Methodology, Investigation, Software, Validation, Writing - Original Draft Preparation; CIM: Methodology, Software, Writing - Review \& Editing; AJP: Methodology, Writing - Original Draft Preparation; NFA: Methodology, Software, Validation, Writing - Original Draft Preparation; DF: Methodology, Software; JC: Validation, Writing - Review \& Editing; DP: Validation, Writing - Review \& Editing; IDB: Validation, Writing - Review & Editing; DA: Software, Validation, Writing - Review \& Editing; AC: Writing - Review \& Editing; AGS: Software, Validation, Writing - Review \& Editing; RC: Software, Validation; DD: Software, Validation; LSK: Validation, Writing - Review \& Editing; AB: Methodology, Writing - Original Draft Preparation; IKM: Software, Validation, Writing - Review \& Editing; TB: Software, Validation, Writing - Review \& Editing; AJ: Software, Validation, Writing - Review \& Editing; TRC: Validation, Writing - Review \& Editing; SMN: Validation, Writing - Review \& Editing; AAW: Writing - Review \& Editing; PEO: Writing - Review \& Editing; GHT: Writing - Review \& Editing; SHT: Writing - Review \& Editing; ARR: Writing - Review \& Editing; BA: Writing - Review \& Editing; DAM: Writing - Review \& Editing; EH: Writing - Review \& Editing; WWLH: Writing - Review \& Editing; ATRV: Writing - Review \& Editing; DRM: Conceptualization, Methodology, Visualization, Writing - Review \& Editing, Funding Acquisition.

\section{Acknowledgements}

The authors would like to thank the US Center for Disease Control and Prevention’s Technical Outreach and Assistance for States Team (TOAST) for their feedback, support, and assistance in disseminating the PHA4GE specification package among US public health networks.