\mbox{}\\
\vspace{8cm}

This chapter is a reproduction of the following publication:

E. J. Griffiths, R. E. Timme, C. I. Mendes, A. J. Page, N. Alikhan, D. Fornika, F. Maguire, J. Campos, D. Park, I. B. Olawoye, P. E. Oluniyi, D. Anderson, A. Christoffels, A. G. da Silva, R. Cameron, D. Dooley, L. S. Katz, A. Black, I. Karsch-Mizrachi, T. Barrett, A. Johnston, T. R. Connor, S. M. Nicholls, A. A. Witney, G. H. Tyson, S. H. Tausch, A. R. Raphenya, B. Alcock, D. M. Aanensen, E. Hodcroft, W. W. L. Hsiao, A. T. R. Vasconcelos, D. R. MacCannell on behalf of the Public Health Alliance for Genomic Epidemiology (PHA4GE) consortium.
Future-proofing and maximizing the utility of metadata: The PHA4GE SARS-CoV-2 contextual data specification package. GigaScience, Volume 11, giac003 (2022). DOI: \url{https://doi.org/10.1093/gigascience/giac003}

The supplementary information referred throughout the text can be consulted in this chapter before the section of references. 

TO CONTINUEEEEE

\cleardoublepage 

\begin{center}
\large
\textbf{Future-proofing and maximizing the utility of metadata: The PHA4GE SARS-CoV-2 contextual data specification package}
\end{center}

Emma J. Griffiths$^1$, 
Ruth E. Timme$^2$,
Catarina I. Mende$^3$,
Andrew J Page$^4$,
Nabil-Fareed Alikha$^4$,
Dan Fornika$^5$,
Finlay Maguire$^6$,
Josefina Campos$^7$,
Daniel Park$^8$,
Idowu B. Olawoy$^{9,10}$,
Paul E. Oluniy$^{9,10}$,
Dominique Anderson$^{11}$,
Alan Christoffel$^{11}$,
Anders Gonçalves da Silva$^{12}$,
Rhiannon Cameron$^1$,
Damion Dooley$^1$,
Lee S. Katz$^{13}$,
Allison Black$^{14}$,
Ilene Karsch-Mizrach$^{15}$,
Tanya Barret$^{15}$,
Anjanette Johnston$^{15}$,
Thomas R. Connor$^{16,17}$,
Samuel M. Nicholls$^{18}$,
Adam A. Witney$^{19}$,
Gregory H. Tyson$^{20}$,
Simon H. Tausch$^{21}$,
Amogelang R. Raphenya$^{22}$,
Brian Alcock$^{22}$,
David M. Aanensen$^{23,24}$,
Emma Hodcrof$^{25}$,
William W. L. Hsiao$^{1,5,26}$,
Ana Tereza R Vasconcelos$^{27}$,
Duncan R MacCannel$^{28}$,
on behalf of the Public Health Alliance for Genomic Epidemiology (PHA4GE) consortium

$^1$  Faculty of Health Sciences, Simon Fraser University, Burnaby, British Columbia,Canada;

$^2$ Center for Food Safety and Applied Nutrition, U.S. Food and Drug Administration, College Park, MD,USA; 

$^3$ Instituto de Microbiologia, Instituto de Medicina Molecular, Faculdade de Medicina, Universidade de Lisboa, Portugal;

$^4$ Quadram Institute Bioscience, Norwich, Norfolk, UK;

$^5$ BC Centre for Disease Control Public Health Laboratory, Vancouver, Canada;

$^6$ Faculty of Computer Science, Dalhousie University, Halifax, Canada;

$^7$ INEI-ANLIS "Dr Carlos G. Malbrán", Buenos Aires, Argentina;

$^8$ The Broad Institute of MIT and Harvard, Cambridge, MA, USA;

$^9$ African Center of Excellence for Genomics of Infectious Diseases (ACEGID), Redeemer's University, Ede, Osun State, Nigeria;

$^{10}$ Department of Biological Sciences, College of Natural Sciences, Redeemer's University, Ede, Osun State, Nigeria;

$^{11}$ South African Medical Research Council Bioinformatics Unit, South African National Bioinformatics Institute, University of the Western Cape, Bellville, South Africa;

$^{12}$  Microbiological Diagnostic Unit Public Health Laboratory, The Peter Doherty Institute for Infection and Immunity, The University of Melbourne, Melbourne, Victoria, Australia;

$^{13}$ Center for Food Safety, University of Georgia, Georgia, USA;

$^{14}$ Department of Epidemiology, University of Washington, Washington, USA;

$^{15}$ National Center for Biotechnology Information, National Library of Medicine, National Institutes of Health, Bethesda, MD, USA;

$^{16}$ Organisms and Environment Division, School of Biosciences, Cardiff University, Cardiff, Wales, UK;

$^{17}$ Public Health Wales, University Hospital of Wales, Cardiff, UK;

$^{18}$ University of Birmingham, Birmingham, UK;

$^{19}$ Institute for Infection and Immunity, St George’s, University of London, London, UK;

$^{20}$ Center for Veterinary Medicine, U.S. Food and Drug Administration, Laurel, Maryland, USA;

$^{21}$ Department of Biological Safety, German Federal Institute for Risk Assessment, Berlin, Germany;

$^{22}$ Department of Biochemistry and Biomedical Sciences and the Michael G. DeGroote Institute forInfectious Disease Research, McMaster University, Hamilton, Ontario, Canada;

$^{23}$ Centre for Genomic Pathogen Surveillance, Wellcome Genome Campus, Cambridge, UK;

$^{24}$ The Big Data Institute, Li Ka Shing Centre for Health Information and Discovery, Nuffield Department of Medicine, University of Oxford, Oxford, UK;

$^{25}$ Biozentrum, University of Basel, Basel, Switzerland \& Swiss Institute of Bioinformatics, Lausanne,Switzerland;

$^{26}$Department of Pathology and Laboratory Medicine, University of British Columbia, Vancouver, Canada;

$^{27}$ Bioinformatics Laboratory National Laboratory of Scientific Computation LNCC/MCTI, Rio de Janeiro, Brazil;

$^{28}$ National Center for Emerging and Zoonotic Infectious Diseases, Centers for Disease Control and Prevention, Georgia, USA

\section{Abstract} \label{sec:ch7_abstract}

The Public Health Alliance for Genomic Epidemiology (PHA4GE) (https://pha4ge.org) is a global coalition that is actively working to establish consensus standards, document and share best practices, improve the availability of critical bioinformatics tools and resources, and advocate for greater openness, interoperability, accessibility and reproducibility in public health microbial bioinformatics. In the face of the current pandemic, PHA4GE has identified a need for a fit-for-purpose, open-source SARS-CoV-2 contextual data standard. As such, we have developed a SARS-CoV-2 contextual data specification package based on harmonisable, publicly available community standards. The specification can be implemented via a collection template, as well as an array of protocols and tools to support both the harmonisation and submission of sequence data and contextual information to public biorepositories. Well-structured, rich contextual data adds value, promotes reuse, and enables aggregation and integration of disparate data sets. Adoption of the proposed standard and practices will better enable interoperability between datasets and systems, improve the consistency and utility of generated data, and ultimately facilitate novel insights and discoveries in SARS-CoV-2 and COVID-19. The package is now supported by the National Center for Biotechnology (NCBI)'s BioSample database.

\section{Findings} \label{sec:ch7_findings}

\subsection{The importance of contextual data for interpreting SARS-CoV-2 sequences}

First identified in late 2019 in Wuhan, China, the SARS-CoV-2 virus has now spread to virtually every country and territory in the world, resulting in millions of confirmed cases, and deaths, globally \cite{world_health_organization_coronavirus_nodate, dong_interactive_2020}. Understanding, monitoring and preventing transmission, as well as the development of vaccines and effective therapeutic options, have been primary goals of the public health response to SARS-CoV-2. 

Tracking the spread and evolution of the virus at global, national and local scales has been aided by the analysis of viral genome sequence data alongside SARS-CoV-2 epidemiology. Large scale sequencing efforts are often formalised as consortia across the world, including the COG-UK in the UK \cite{covid-19_genomics_uk_cog-uk_consortiumcontactcogconsortiumuk_integrated_2020}, SPHERES in the USA \cite{cdc_cases_2020}, CanCOGeN in Canada \cite{cancogen_genome_canada_cancogen_nodate}, Latin American Genomics SARS-CoV-2 Network \cite{pan_american_health_organization_laboratory_nodate, candido_evolution_2020}, 2019nCoVR in China \cite{zhao_2019_2020}, the South Africa NGS Genomic Surveillance Network \cite{network_for_genomic_surveillance_south_africa_ngs-sa_nodate}, AusTrakka in Australia and New Zealand \cite{communicable_diseases_genomics_network_austrakka_nodate}, and INSACOG in India \cite{government_of_india_indian_nodate}. In addition to these initiatives, many agencies, universities and hospital laboratories around the world are also sequencing and sharing sequence data at an unprecedented pace. Deposition of these sequences into public repositories such as the Global Initiative on Sharing All Influenza Data (GISAID) and the International Nucleotide Sequence Database Collaboration (INSDC) has enabled rapid global sharing of data \cite{shu_gisaid_2017,karsch-mizrachi_international_2018}. At the time of writing, 174 countries had undertaken open sequencing initiatives (GISAID accessed 2021-06-23) depositing 2,057,675 sequences which are being reused and analysed on a massive scale. The open data sharing paradigm has had tremendous success in the genomic epidemiology of foodborne pathogens \cite{allard_practical_2016,kubota_pulsenet_2019}, and has the potential to reveal a deeper understanding of SARS-CoV-2 origin, pathogenicity, and basic biology when submissions from environmental samples and wild hosts are included alongside human clinical samples \cite{cook_integrating_2020}.

SARS-CoV-2 sequencing, analysis, and open sharing have played a crucial role in a number of developments during the pandemic, such as dispelling misinformation about the origins of the virus \cite{andersen_proximal_2020}, the identification and surveillance of variants of concern \cite{gupta_will_2021}, \cite{public_health_england_sars-cov-2_nodate}, the improvement of diagnostic performance and rapid testing \cite{los_alamos_national_laboratory_silico_nodate, kuchinski_mutations_2022, ganguli_rapid_2020}, and the development of vaccines which are currently being distributed in the largest global vaccination program the world has ever seen \cite{world_health_organization_covid-19_nodate}. Viral genomic sequences are also being used to understand transmission and reinfection events \cite{tillett_genomic_2021} as well to monitor the prevalence and diversity of lineages during different exposure events and in different settings e.g. animal reservoirs \cite{oude_munnink_transmission_2021}, long-term care facilities \cite{lai_covid-19_2020, aggarwal_role_2020, murti_investigation_2021}, healthcare and other work sites \cite{dyal_covid-19_2020, gunther_sarscov2_2020, taylor_serial_2020, loconsole_investigation_2021, frampton_genomic_2021}, conferences and other public gatherings \cite{da_silva_filipe_genomic_2021}, as well before and after public health responses (e.g. border controls and travel restrictions, lockdowns and quarantines, vaccination, etc.), through successive waves of infections \cite{oude_munnink_rapid_2020, du_plessis_establishment_2021, githinji_tracking_2020, meredith_rapid_2020, zhang_analysis_2020, long_molecular_2020, geoghegan_genomic_2020, seemann_tracking_2020, mclaughlin_early_2021, fauver_coast--coast_2020, knock_key_2021, lane_genomics-informed_2021}. However, it is critical to note that public health sequence data is of limited value without accompanying contextual metadata. 

Contextual data consists of sample metadata (e.g., collection date, sample type, geographical location of sample collection), as well as laboratory (e.g., date and location testing, cycle threshold (CT) values), clinical outcomes (e.g., hospitalization, death, recovery), epidemiological (e.g., age, gender, exposures, vaccination status) and methods (e.g., sampling, sequencing, bioinformatics) that enable the interpretation sequence data (e.g., previous examples). High-quality contextual data is also crucial for quality control. For example, detecting systematic batch effect errors related to certain sequencing centres and methods can help evaluate which variants represent real, circulating viruses, as opposed to artifacts of sample handling or sequencing which may arise due to different aspects experimental design, laboratory procedures, bioinformatics processing, and applied quality control thresholds \cite{de_maio_issues_2020, rayko_quality_2020, poon_recurrent_2005}. 

Good data stewardship practices are not only critical for auditability and reproducibility, but for posterity - documenting critical information about samples, methods, risk factors and outcomes etc., can help future-proof information used to build a roadmap for dealing with future public health crises. Contextual data, however, is often collected on a project-specific basis according to local needs and reporting requirements which results in the collection of different data types at different levels of granularity, with different meanings and implicit bias of variables and attributes. Furthermore, the information is often collected as free text, or if structured, according to organization or initiative-specific data dictionaries, using different fields, terms, formats, abbreviations, and jargon.

The variability in the way information is encoded in private databases tends to propagate to public repositories, which makes the information more difficult to interpret and to use. There are different existing standards that can be used to structure contextual data, like minimum information checklists (MIxS \cite{yilmaz_minimum_2011}, MIGS \cite{field_minimum_2008}, the NIAID/BRC Project and Sample Application Standard \cite{dugan_standardized_2014}) and various interoperable ontologies (OBO Foundry \cite{smith_obo_2007}), which make information easier to aggregate and reuse for different types of analyses. However, these attribute packages and metadata standards developed by different organizations are usually scoped to cover as many use cases and pathogens as possible, and as such, can include fields of information not applicable to SARS-CoV-2, or that may be subject to privacy concerns, or exclude fields commonly used in public health surveillance and investigations. As different types of contextual data are subject to different ethical, practical and privacy concerns, not all components of existing standards are immediately or widely collectable and shareable. As a result, the range of generic metadata standards being applied to SARS-CoV-2 data presents challenges for data harmonization \cite{schriml_covid-19_2020} and analysis critical for fighting the disease and ending the pandemic. 

In light of these challenges, PHA4GE has identified a need for a fit-for-purpose, open-source SARS-CoV-2 contextual data specification which can be used to consistently structure information as part of good data management practices and for data sharing with trusted partners and/or public repositories. The specification was developed by consensus among domain experts, and incorporates existing community standards with an emphasis on SARS-CoV-2 public health needs and ensuring privacy while maximizing information content and interoperability across datasets and databases to better enable analyses to fight COVID-19. The specification package also contains a number of accompanying materials such as standard operating procedures, tools, a reference guide, and repository submission protocols (protocols.io) to help put the standard into practice.  

\subsection{SARS-CoV-2 Contextual Data Specification: The Framework}

The purpose of the PHA4GE SARS-CoV-2 specification is to provide a mechanism for consistent structure, collection and formatting of fields and values containing SARS-CoV-2 contextual data pertaining to clinical, animal, and environmental samples. We emphasize that the purpose of this specification is not to force data sharing, but rather to provide a framework to structure data consistently across disparate laboratory and epidemiological databases so that they can be harmonized for different uses (Figure \ref{fig:chap7_figure_1}). Data sharing is just one use case and can involve sharing between divisions within a single agency, sharing between partners based on memorandums of understanding, or submission to public repositories. 

The PHA4GE SARS-CoV-2 contextual data specification was created through broad consultation with representatives from public health laboratories, research institutes and universities in 11 countries (Argentina, Australia, Brazil, Canada, Germany, Nigeria, Portugal, South Africa, Switzerland, the United Kingdom, the United States of America) who are involved with the SARS-CoV-2 genome sequencing and analysis efforts at various scales. Based on this consultation and consensus, the specification contains different fields covering a wide array of data types described in Box 1 (Figure \ref{fig:chap7_figure_1}). The specification attempts to harmonize different data standards (INSDC, GISAID, MIxS, MIGS, Sample Application Standard) by reusing fields or mapping to fields, as much as possible. As PHA4GE embraces FAIR data stewardship principles (Findability, Accessibility, Interoperability and Reuse of digital assets), we strived to implement FAIR principles in the design and implementation of the specification for data management and data sharing. At their core, these principles emphasize machine-actionability and consistency of data, and are critical for dealing with the volume and complexity of genomic sequence and contextual data. Principles of FAIR data stewardship that have been implemented include improving machine-actionability of data by using a formal, accessible, shared, and broadly applicable language for knowledge representation, reusing existing standards and ontology-based vocabulary to increase interoperability, providing a data usage license, capturing data provenance, and making all resources open, free and widely accessible.

\begin{figure*}[h!]
\centering
\includegraphics[width=\textwidth]{figures/chapter 7/giac003fig1.jpeg}
\caption{Contextual data flow. Contextual data can be captured and structured using the PHA4GE specification so that they can be more easily harmonized across different data sources and providers. Different subsets of the harmonized data can be (i) shared with public repositories, e.g., GISAID and INSDC; (ii) shared with trusted partners, e.g., national sequencing consortia, public health partners; and (iii) kept private and retained locally with the potential for sharing in the future for particular surveillance or research activities. While fields have been colour-coded in the template to indicate whether they are considered “required,” “strongly recommended,” or “optional,” how the specification is implemented and whether any of the data are shared is ultimately at the discretion of the user. Box 1 describes the information types covered in the full specification.}
\label{fig:chap7_figure_1}
\end{figure*}

The versioned specification is available as a contextual data collection template (.xlsx) and in machine-amenable JSON format from GitHub (v 3.0.0 - https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification) \cite{public_health_alliance_for_genomic_epidemiology_sars-cov-2-contextual-data-specification_nodate}. The collection template also offers standardized terms for a number of fields in the form of pick lists. The fields are colour-coded to indicate required (yellow), strongly recommended (purple) or optional status (white). Fields useful for surveillance were prioritized as required. Formats for data elements like dates are also prescribed according to international standards (e.g., dates should be formatted according to ISO8601).

The template is also supported by several materials such as term and field-level Reference Guides (available as tabs in the collection template Excel workbook), which provides definitions, data entry guidance and examples of usage \cite{public_health_alliance_for_genomic_epidemiology_sars-cov-2-contextual-data-specification_nodate}. The field-level Reference Guide also provides mapping of PHA4GE fields to existing contextual data standards, highlighting public health and SARS-CoV-2-specific fields that were missing as well as fields in those other standards that were considered out of scope. 

The Open Biological and Biomedical Ontology (OBO) Foundry is a community of researchers that use a prescribed set of principles and practices to develop a wide range of interoperable ontologies focused on the life sciences \cite{the_obo_foundry_obo_nodate}. Fields and terms in the specification have been mapped to existing OBO Foundry ontology terms, and where required, new ontology terms have been developed and are being made available in different application and domain-specific ontologies within The Foundry (see Table \ref{tab:ch7_table1} for a list of source ontologies). As of version 3.0.0 and beyond, terms in pick lists provided in the collection template are presented with corresponding ontology identifiers in the format “Label [ontology ID]” e.g., Blood [UBERON:0000178]. Axioms and additional cross references to ontologies and existing standards are actively being developed in collaboration with community developers. We anticipate that our contributions to these freely available, open-source resources will be of use to the COVID-19 research community.

\begin{table}[h!]
\caption{Ontologies implemented in the PHA4GE SARS-CoV-2 specification.}
\label{tab:ch7_table1}
\resizebox{\textwidth}{!}{%
\begin{tabular}{@{}ll@{}}
\toprule
Ontology$^1$                                    & Link                                           \\ \midrule
BRENDA Tissue Ontology (BTO)                 & https://obofoundry.org/ontology/bto.html       \\
Cell Line Ontology (CLO)                     & https://obofoundry.org/ontology/clo.html       \\
Environmental conditions,   treatments and exposures ontology (ECTO) & https://obofoundry.org/ontology/ecto.html   \\
Environment Ontology (ENVO)                  & https://obofoundry.org/ontology/envo.html      \\
Food Ontology (FoodOn)                       & https://obofoundry.org/ontology/foodon.html    \\
Gazetteer Ontology (GAZ)                     & https://obofoundry.org/ontology/gaz.html       \\
Gender, Sex, and Sexual Orientation Ontology (GSSO)                  & https://obofoundry.org/ontology/gsso.html   \\
Genomic Epidemiology Ontology (GenEpiO)      & https://obofoundry.org/ontology/genepio.html   \\
Genomics Cohorts Knowledge Ontology (GECKO)  & https://obofoundry.org/ontology/gecko.html     \\
Human Disease Ontology (DOID)                & https://obofoundry.org/ontology/doid.html      \\
Human Phenotype Ontology (HP)                & https://obofoundry.org/ontology/hp.html        \\
Mammalian Phenotype Ontology (MP)            & https://obofoundry.org/ontology/mp.html        \\
Measurement Method Ontology (MMO)            & https://obofoundry.org/ontology/mmo.html       \\
Mondo Disease Ontology (MONDO)               & https://obofoundry.org/ontology/mondo.html     \\
Mouse Pathology Ontology (MPATH)             & https://obofoundry.org/ontology/mpath.html     \\
National Cancer Institute Thesaurus (NCIT)   & https://obofoundry.org/ontology/ncit.html      \\
NCBI Taxonomy Ontology (NCBITaxon)           & https://obofoundry.org/ontology/ncbitaxon.html \\
Neuro Behaviour Ontology (NBO)               & https://obofoundry.org/ontology/nbo.html       \\
Ontology for Biomedical Investigations (OBI) & https://obofoundry.org/ontology/obi.html       \\
Ontology of Medically Related Social Entities (OMRSE)                & https://obofoundry.org/ontology/omrse.html  \\
Population and Community Ontology (PCO)      & https://obofoundry.org/ontology/pco.html       \\
UBERON Multi-species Anatomy Ontology (UBERON)                       & https://obofoundry.org/ontology/uberon.html \\
Unit Ontology (UO)                           & https://obofoundry.org/ontology/uo.html        \\
Vaccine Ontology (VO)                        & https://obofoundry.org/ontology/vo.html        \\ \bottomrule
\end{tabular}%
}
\item $^1$ Vocabulary for fields and terms in the specification have been sourced or mapped to OBO Foundry domain and application ontologies, which are highlighted in this list. New fields and terms for which there were no existing equivalents have been developed and submitted to these ontologies, expanding these community resources
\end{table}

Protocols have also been created and are openly available on protocols.io \cite{public_health_alliance_for_genomic_epidemiology_pha4ge_nodate}, including a curation Standard Operating Procedure (SOP) containing instructions for using the collection template as well as guidance for a number of privacy and practical concerns. A series of versioned SARS-CoV-2 sequence and contextual data submission protocols and accompanying instructional videos for how to prepare submissions and navigate through the various submission portals for GISAID, NCBI and EMBL-EBI are also provided.

A mapping file indicating which PHA4GE fields correspond to contextual data elements recommended by the World Health Organization has been provided to help data providers comply with international guidance \cite{world_health_organization_guidance_nodate}. This mapping file also includes tabs indicating which PHA4GE fields correspond to those found in different repository submission forms to facilitate data transformations for submissions. Such transformations can be automated using a contextual data harmonization application called the DataHarmonizer \cite{hsiao_public_health_bioinformatics_lab_dataharmonizer_2022}. PHA4GE has worked with the developers of the DataHarmonizer to offer the PHA4GE standard as a template in the tool (Gill et al, in preparation). Users can standardize and validate entered data and export it as GISAID and NCBI-ready submission forms (BioSample, SRA, GenBank and GenBank source modifier forms). It should be noted that other excellent contextual data transformation tools have been developed by the community, such as METAGENOTE, multiSub, and a GISAID-to-ENA conversion script \cite{noauthor_metagenote_nodate, haeussler_multisub_2022, noauthor_ena-content-dataflowscriptsgisaid_to_ena_nodate}.

A table outlining the different specification package materials can be found in Table \ref{tab:ch7_table2}.


\begin{longtable}{@{}lll@{}}
\caption{Resources that form the PHA4GE SARS-CoV-2 contextual data specification package}
\label{tab:ch7_table2}\\
Resource1 &
  Description &
  Link \\
\endfirsthead
%
\endhead
%
Collection   template and controlled vocabulary pick lists &
  Spreadsheet-based   collection form containing different fields (identifiers and accessions,   sample collection and processing, host information, host exposure,   vaccination and reinfection information, lineage and variant information,   sequencing, bioinformatics and QC metrics, diagnostic testing information,   author acknowledgements). Fields are colour-coded to indicate required,   recommended or optional status. Many fields offer pick lists of controlled   vocabulary. Vocabulary lists are also available in a separate tab. &
  https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/raw/master/PHA4GE\%20SARS-CoV-2\%20Contextual\%20Data\%20Template.xlsx \\
Reference   guides &
  Field and   term definitions, guidance, and examples are provided as separate tabs in the   collection template .xlsx file (see Term Reference Guide and Field Reference   Guide). &
  https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/raw/master/PHA4GE\%20SARS-CoV-2\%20Contextual\%20Data\%20Template.xlsx \\
Curation   protocol on protocols.io &
  Step-by-step   instructions for using the collection template are provided in a standard   operating procedure (SOP). Ethical, practical, and privacy considerations are   also discussed. Examples and instructions for structuring sample descriptions   as well as sourcing additional standardized terms (outside those provided in   pick lists) are also discussed. &
  dx.doi.org/10.17504/protocols.io.btpznmp6 \\
Mapping   file of PHA4GE fields to metadata standards &
  PHA4GE   fields are mapped to existing metadata standards such as the Sample   Application Standard, MIxS 5.0, and the MIGS Virus Host-associated attribute   package. Mappings are available in the Reference guide tab. Mappings   highlight which fields of these standards are considered useful for SARS-CoV-2   public health surveillance and investigations, and which fields are   considered out of scope. &
  https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/raw/master/PHA4GE\%20SARS-CoV-2\%20Contextual\%20Data\%20Template.xlsx \\
Mapping   of PHA4GE fields to WHO metadata recommendations &
  PHA4GE   fields are mapped to corresponding contextual data elements recommended by   the World Health Organization. &
  https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/raw/master/PHA4GE\%20to\%20Sequence\%20Repository\%20Field\%20Mappings.xlsx \\
Mapping   file of PHA4GE fields to EMBL-EBI, NCBI and GISAID submission requirements &
  Many   PHA4GE fields have been sourced from public repository submission   requirements. The different repositories have different requirements and   field names. Repository submission fields have been mapped to PHA4GE fields   to demonstrate equivalencies and divergences. &
  https://github.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/raw/master/PHA4GE\%20to\%20Sequence\%20Repository\%20Field\%20Mappings.xlsx \\
Data submission protocol (NCBI) on protocols.io &
  The   SARS-CoV-2 submission protocol for NCBI provides step-by-step instructions   and recommendations aimed at improving interoperability and consistency of   submitted data. &
  dx.doi.org/10.17504/protocols.io.bui7nuhn \\
Data submission protocol (EMBL-EBI) on protocols.io &
  The   SARS-CoV-2 submission protocol for ENA provides step-by-step instructions and   recommendations aimed at improving interoperability and consistency of   submitted data. &
  dx.doi.org/10.17504/protocols.io.buqnnvve \\
Data   submission protocol (GISAID) on protocols.io &
  The   SARS-CoV-2 submission protocol for GISAID provides step-by-step instructions   and recommendations aimed at improving interoperability and consistency of submitted   data. &
  dx.doi.org/10.17504/protocols.io.bumknu4w \\
JSON   structure of PHA4GE specification &
  A JSON   structure of the PHA4GE specification has been provided for easier   integration into software applications. &
  https://raw.githubusercontent.com/pha4ge/SARS-CoV-2-Contextual-Data-Specification/master/PHA4GE\_SARS-CoV-2\_Contextual\_Data\_Schema.json \\
PHA4GE   template in the DataHarmonizer &
  Javascript   application enabling standardized data entry, validation and export of   contextual data as submission-ready forms for GISAID and NCBI. The SOP for   using the software can be found at   https://github.com/Public-Health-Bioinformatics/DataHarmonizer/wiki/PHA4GE-SARS-CoV-2-Template &
  https://github.com/Public-Health-Bioinformatics/DataHarmonizer/releases
  
\item $^1$ There are a number of resources that form the PHA4GE SARS-CoV-2 contextual data specification package which are described in the table. The package has been compiled to support user implementation and data sharing, with integration into workflows and new software applications in mind.
\end{longtable}


