\mbox{}\\
\vspace{8cm}

This chapter is a reproduction of the following publication:

E. J. Griffiths, R. E. Timme, C. I. Mendes, A. J. Page, N. Alikhan, D. Fornika, F. Maguire, J. Campos, D. Park, I. B. Olawoye, P. E. Oluniyi, D. Anderson, A. Christoffels, A. G. da Silva, R. Cameron, D. Dooley, L. S. Katz, A. Black, I. Karsch-Mizrachi, T. Barrett, A. Johnston, T. R. Connor, S. M. Nicholls, A. A. Witney, G. H. Tyson, S. H. Tausch, A. R. Raphenya, B. Alcock, D. M. Aanensen, E. Hodcroft, W. W. L. Hsiao, A. T. R. Vasconcelos, D. R. MacCannell on behalf of the Public Health Alliance for Genomic Epidemiology (PHA4GE) consortium.
Future-proofing and maximizing the utility of metadata: The PHA4GE SARS-CoV-2 contextual data specification package. GigaScience, Volume 11, giac003 (2022). DOI: \url{https://doi.org/10.1093/gigascience/giac003}

The supplementary information referred throughout the text can be consulted in this chapter before the section of references. 

TO CONTINUEEEEE

\cleardoublepage 

\begin{center}
\large
\textbf{Future-proofing and maximizing the utility of metadata: The PHA4GE SARS-CoV-2 contextual data specification package}
\end{center}

Emma J. Griffiths$^1$, 
Ruth E. Timme$^2$,
Catarina I. Mende$^3$,
Andrew J Page$^4$,
Nabil-Fareed Alikha$^4$,
Dan Fornika$^5$,
Finlay Maguire$^6$,
Josefina Campos$^7$,
Daniel Park$^8$,
Idowu B. Olawoy$^{9,10}$,
Paul E. Oluniy$^{9,10}$,
Dominique Anderson$^{11}$,
Alan Christoffel$^{11}$,
Anders Gonçalves da Silva$^{12}$,
Rhiannon Cameron$^1$,
Damion Dooley$^1$,
Lee S. Katz$^{13}$,
Allison Black$^{14}$,
Ilene Karsch-Mizrach$^{15}$,
Tanya Barret$^{15}$,
Anjanette Johnston$^{15}$,
Thomas R. Connor$^{16,17}$,
Samuel M. Nicholls$^{18}$,
Adam A. Witney$^{19}$,
Gregory H. Tyson$^{20}$,
Simon H. Tausch$^{21}$,
Amogelang R. Raphenya$^{22}$,
Brian Alcock$^{22}$,
David M. Aanensen$^{23,24}$,
Emma Hodcrof$^{25}$,
William W. L. Hsiao$^{1,5,26}$,
Ana Tereza R Vasconcelos$^{27}$,
Duncan R MacCannel$^{28}$,
on behalf of the Public Health Alliance for Genomic Epidemiology (PHA4GE) consortium

$^1$  Faculty of Health Sciences, Simon Fraser University, Burnaby, British Columbia,Canada;

$^2$ Center for Food Safety and Applied Nutrition, U.S. Food and Drug Administration, College Park, MD,USA; 

$^3$ Instituto de Microbiologia, Instituto de Medicina Molecular, Faculdade de Medicina, Universidade de Lisboa, Portugal;

$^4$ Quadram Institute Bioscience, Norwich, Norfolk, UK;

$^5$ BC Centre for Disease Control Public Health Laboratory, Vancouver, Canada;

$^6$ Faculty of Computer Science, Dalhousie University, Halifax, Canada;

$^7$ INEI-ANLIS "Dr Carlos G. Malbrán", Buenos Aires, Argentina;

$^8$ The Broad Institute of MIT and Harvard, Cambridge, MA, USA;

$^9$ African Center of Excellence for Genomics of Infectious Diseases (ACEGID), Redeemer's University, Ede, Osun State, Nigeria;

$^{10}$ Department of Biological Sciences, College of Natural Sciences, Redeemer's University, Ede, Osun State, Nigeria;

$^{11}$ South African Medical Research Council Bioinformatics Unit, South African National Bioinformatics Institute, University of the Western Cape, Bellville, South Africa;

$^{12}$  Microbiological Diagnostic Unit Public Health Laboratory, The Peter Doherty Institute for Infection and Immunity, The University of Melbourne, Melbourne, Victoria, Australia;

$^{13}$ Center for Food Safety, University of Georgia, Georgia, USA;

$^{14}$ Department of Epidemiology, University of Washington, Washington, USA;

$^{15}$ National Center for Biotechnology Information, National Library of Medicine, National Institutes of Health, Bethesda, MD, USA;

$^{16}$ Organisms and Environment Division, School of Biosciences, Cardiff University, Cardiff, Wales, UK;

$^{17}$ Public Health Wales, University Hospital of Wales, Cardiff, UK;

$^{18}$ University of Birmingham, Birmingham, UK;

$^{19}$ Institute for Infection and Immunity, St George’s, University of London, London, UK;

$^{20}$ Center for Veterinary Medicine, U.S. Food and Drug Administration, Laurel, Maryland, USA;

$^{21}$ Department of Biological Safety, German Federal Institute for Risk Assessment, Berlin, Germany;

$^{22}$ Department of Biochemistry and Biomedical Sciences and the Michael G. DeGroote Institute forInfectious Disease Research, McMaster University, Hamilton, Ontario, Canada;

$^{23}$ Centre for Genomic Pathogen Surveillance, Wellcome Genome Campus, Cambridge, UK;

$^{24}$ The Big Data Institute, Li Ka Shing Centre for Health Information and Discovery, Nuffield Department of Medicine, University of Oxford, Oxford, UK;

$^{25}$ Biozentrum, University of Basel, Basel, Switzerland \& Swiss Institute of Bioinformatics, Lausanne,Switzerland;

$^{26}$Department of Pathology and Laboratory Medicine, University of British Columbia, Vancouver, Canada;

$^{27}$ Bioinformatics Laboratory National Laboratory of Scientific Computation LNCC/MCTI, Rio de Janeiro, Brazil;

$^{28}$ National Center for Emerging and Zoonotic Infectious Diseases, Centers for Disease Control and Prevention, Georgia, USA

\section{Abstract} \label{sec:ch7_abstract}

The Public Health Alliance for Genomic Epidemiology (PHA4GE) (https://pha4ge.org) is a global coalition that is actively working to establish consensus standards, document and share best practices, improve the availability of critical bioinformatics tools and resources, and advocate for greater openness, interoperability, accessibility and reproducibility in public health microbial bioinformatics. In the face of the current pandemic, PHA4GE has identified a need for a fit-for-purpose, open-source SARS-CoV-2 contextual data standard. As such, we have developed a SARS-CoV-2 contextual data specification package based on harmonisable, publicly available community standards. The specification can be implemented via a collection template, as well as an array of protocols and tools to support both the harmonisation and submission of sequence data and contextual information to public biorepositories. Well-structured, rich contextual data adds value, promotes reuse, and enables aggregation and integration of disparate data sets. Adoption of the proposed standard and practices will better enable interoperability between datasets and systems, improve the consistency and utility of generated data, and ultimately facilitate novel insights and discoveries in SARS-CoV-2 and COVID-19. The package is now supported by the National Center for Biotechnology (NCBI)'s BioSample database.

\section{Findings} \label{sec:ch7_findings}

\subsection{The importance of contextual data for interpreting SARS-CoV-2 sequences}

First identified in late 2019 in Wuhan, China, the SARS-CoV-2 virus has now spread to virtually every country and territory in the world, resulting in millions of confirmed cases, and deaths, globally \cite{world_health_organization_coronavirus_nodate, dong_interactive_2020}. Understanding, monitoring and preventing transmission, as well as the development of vaccines and effective therapeutic options, have been primary goals of the public health response to SARS-CoV-2. 

Tracking the spread and evolution of the virus at global, national and local scales has been aided by the analysis of viral genome sequence data alongside SARS-CoV-2 epidemiology. Large scale sequencing efforts are often formalised as consortia across the world, including the COG-UK in the UK \cite{covid-19_genomics_uk_cog-uk_consortiumcontactcogconsortiumuk_integrated_2020}, SPHERES in the USA \cite{cdc_cases_2020}, CanCOGeN in Canada \cite{cancogen_genome_canada_cancogen_nodate}, Latin American Genomics SARS-CoV-2 Network \cite{pan_american_health_organization_laboratory_nodate, candido_evolution_2020}, 2019nCoVR in China \cite{zhao_2019_2020}, the South Africa NGS Genomic Surveillance Network \cite{network_for_genomic_surveillance_south_africa_ngs-sa_nodate}, AusTrakka in Australia and New Zealand \cite{communicable_diseases_genomics_network_austrakka_nodate}, and INSACOG in India \cite{government_of_india_indian_nodate}. In addition to these initiatives, many agencies, universities and hospital laboratories around the world are also sequencing and sharing sequence data at an unprecedented pace. Deposition of these sequences into public repositories such as the Global Initiative on Sharing All Influenza Data (GISAID) and the International Nucleotide Sequence Database Collaboration (INSDC) has enabled rapid global sharing of data \cite{shu_gisaid_2017,karsch-mizrachi_international_2018}. At the time of writing, 174 countries had undertaken open sequencing initiatives (GISAID accessed 2021-06-23) depositing 2,057,675 sequences which are being reused and analysed on a massive scale. The open data sharing paradigm has had tremendous success in the genomic epidemiology of foodborne pathogens \cite{allard_practical_2016,kubota_pulsenet_2019}, and has the potential to reveal a deeper understanding of SARS-CoV-2 origin, pathogenicity, and basic biology when submissions from environmental samples and wild hosts are included alongside human clinical samples \cite{cook_integrating_2020}.

SARS-CoV-2 sequencing, analysis, and open sharing have played a crucial role in a number of developments during the pandemic, such as dispelling misinformation about the origins of the virus \cite{andersen_proximal_2020}, the identification and surveillance of variants of concern \cite{gupta_will_2021}, \cite{public_health_england_sars-cov-2_nodate}, the improvement of diagnostic performance and rapid testing \cite{los_alamos_national_laboratory_silico_nodate, kuchinski_mutations_2022, ganguli_rapid_2020}, and the development of vaccines which are currently being distributed in the largest global vaccination program the world has ever seen \cite{world_health_organization_covid-19_nodate}. Viral genomic sequences are also being used to understand transmission and reinfection events \cite{tillett_genomic_2021} as well to monitor the prevalence and diversity of lineages during different exposure events and in different settings e.g. animal reservoirs \cite{oude_munnink_transmission_2021}, long-term care facilities \cite{lai_covid-19_2020, aggarwal_role_2020, murti_investigation_2021}, healthcare and other work sites \cite{dyal_covid-19_2020, gunther_sarscov2_2020, taylor_serial_2020, loconsole_investigation_2021, frampton_genomic_2021}, conferences and other public gatherings \cite{da_silva_filipe_genomic_2021}, as well before and after public health responses (e.g. border controls and travel restrictions, lockdowns and quarantines, vaccination, etc.), through successive waves of infections \cite{oude_munnink_rapid_2020, du_plessis_establishment_2021, githinji_tracking_2020, meredith_rapid_2020, zhang_analysis_2020, long_molecular_2020, geoghegan_genomic_2020, seemann_tracking_2020, mclaughlin_early_2021, fauver_coast--coast_2020, knock_key_2021, lane_genomics-informed_2021}. However, it is critical to note that public health sequence data is of limited value without accompanying contextual metadata. 

Contextual data consists of sample metadata (e.g., collection date, sample type, geographical location of sample collection), as well as laboratory (e.g., date and location testing, cycle threshold (CT) values), clinical outcomes (e.g., hospitalization, death, recovery), epidemiological (e.g., age, gender, exposures, vaccination status) and methods (e.g., sampling, sequencing, bioinformatics) that enable the interpretation sequence data (e.g., previous examples). High-quality contextual data is also crucial for quality control. For example, detecting systematic batch effect errors related to certain sequencing centres and methods can help evaluate which variants represent real, circulating viruses, as opposed to artifacts of sample handling or sequencing which may arise due to different aspects experimental design, laboratory procedures, bioinformatics processing, and applied quality control thresholds \cite{de_maio_issues_2020, rayko_quality_2020, poon_recurrent_2005}. 

Good data stewardship practices are not only critical for auditability and reproducibility, but for posterity - documenting critical information about samples, methods, risk factors and outcomes etc., can help future-proof information used to build a roadmap for dealing with future public health crises. Contextual data, however, is often collected on a project-specific basis according to local needs and reporting requirements which results in the collection of different data types at different levels of granularity, with different meanings and implicit bias of variables and attributes. Furthermore, the information is often collected as free text, or if structured, according to organization or initiative-specific data dictionaries, using different fields, terms, formats, abbreviations, and jargon.

The variability in the way information is encoded in private databases tends to propagate to public repositories, which makes the information more difficult to interpret and to use. There are different existing standards that can be used to structure contextual data, like minimum information checklists (MIxS \cite{yilmaz_minimum_2011}, MIGS \cite{field_minimum_2008}, the NIAID/BRC Project and Sample Application Standard \cite{dugan_standardized_2014}) and various interoperable ontologies (OBO Foundry \cite{smith_obo_2007}), which make information easier to aggregate and reuse for different types of analyses. However, these attribute packages and metadata standards developed by different organizations are usually scoped to cover as many use cases and pathogens as possible, and as such, can include fields of information not applicable to SARS-CoV-2, or that may be subject to privacy concerns, or exclude fields commonly used in public health surveillance and investigations. As different types of contextual data are subject to different ethical, practical and privacy concerns, not all components of existing standards are immediately or widely collectable and shareable. As a result, the range of generic metadata standards being applied to SARS-CoV-2 data presents challenges for data harmonization \cite{schriml_covid-19_2020} and analysis critical for fighting the disease and ending the pandemic. 

In light of these challenges, PHA4GE has identified a need for a fit-for-purpose, open-source SARS-CoV-2 contextual data specification which can be used to consistently structure information as part of good data management practices and for data sharing with trusted partners and/or public repositories. The specification was developed by consensus among domain experts, and incorporates existing community standards with an emphasis on SARS-CoV-2 public health needs and ensuring privacy while maximizing information content and interoperability across datasets and databases to better enable analyses to fight COVID-19. The specification package also contains a number of accompanying materials such as standard operating procedures, tools, a reference guide, and repository submission protocols (protocols.io) to help put the standard into practice.  

\subsection{SARS-CoV-2 Contextual Data Specification: The Framework}

The purpose of the PHA4GE SARS-CoV-2 specification is to provide a mechanism for consistent structure, collection and formatting of fields and values containing SARS-CoV-2 contextual data pertaining to clinical, animal, and environmental samples. We emphasize that the purpose of this specification is not to force data sharing, but rather to provide a framework to structure data consistently across disparate laboratory and epidemiological databases so that they can be harmonized for different uses (Figure \ref{}). Data sharing is just one use case and can involve sharing between divisions within a single agency, sharing between partners based on memorandums of understanding, or submission to public repositories. 

\begin{figure*}[h!]
\centering
\includegraphics[width=\textwidth]{figures/chapter 4/Figure S1.pdf}
\caption{DEN-IM report tables. a) DEN-IM's quality control report containing information of the number of base-pairs and the number of reads for the analysed samples, the estimated coverage depth before and after mapping, and the percentage of reads in the input data that were trimmed. b) DEN-IM's typing report for 24 CDSs recovered from the metagenomic dataset. The ID contains the CDS contig name, the typing result for serotype-genotype, the values for identity and coverage, and the GenBank ID of the closest reference in the Typing Database containing 161 complete DENV genomes.}
\label{fig:chap4_figure_sup1}
\end{figure*}